services/encoding_service.go

```diff
@@ -129,6 +129,33 @@ func (s *EncodingService) AddIngestStream(encodingID string, name string, inputI
 	return &r, nil
 }
 
+func (s *EncodingService) AddDolbyVisionInputStream(encodingID string, name string, inputID string, videoInputPath string, metadataInputPath string) (*models.DolbyVisionStreamResponse, error) {
+	dolbyVisionStream := struct {
+		Name              string `json:"name"`
+		InputID           string `json:"inputId"`
+		VideoInputPath    string `json:"videoInputPath"`
+		MetadataInputPath string `json:"metadataInputPath,omitempty"`
+	}{
+		Name:              name,
+		InputID:           inputID,
+		VideoInputPath:    videoInputPath,
+		MetadataInputPath: metadataInputPath,
+	}
+
+	b, _ := json.Marshal(dolbyVisionStream)
+	path := EncodingEndpoint + "/" + encodingID + "/" + "input-streams" + "/" + "dolby-vision"
```

Comment:
maybe a bit too much concatenation for a hardcoded second part of the path 🤣  but whatever

Reply:
yep, good point. I copied and pasted from `AddIngestStream` and did not realize that :). Removed the unnecessary concatenation